Automattic/pocket-casts-android
# Adding Deselect Options to the Multiselect Dialog

## Description

This PR adds multi-select options to the FilterEpisodesListFragment, UpNextFragment, ProfileEpisodeListFragment, PodcastFragment and CloudFilesFragment. 
The deselect functions would pop up when the selectedList size is more than 1, because for the first time, the user would not think of deselecting the list. Deselect options are: Deselect all, Deselect all above and Deselect all below. 
I also switched the position of deselecting below/selecting below with the select all/ deselect all so that the selecting below/deselect below text option is below the select all(check the screenshot attached)

Suggestion maybe: We could improve the icons for deselecting probably instead of the tick checkbox, we can make something like this [X]. 

Fixes # @CookieyedCodes niggles 

## Testing Instructions

1. Tap on Podcast
2. Scroll to the episodes and long press any podcast(this would trigger multi-select)
3. Tap on the selected option(checkout the dialog that pops out(select all above/select all/ select all below)
4. tap on a second multiselect option
5. Long press to check out the new dialog(Deselect all above/ Deselect all/ Deselect all below)
6. Have fun testing the dialog options 

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/95022986/e6e46257-a41b-457c-a406-d279210c2b95

|   Deselect Dialog    |   Select Dialog  | 
|---	|--- 
 ![deselect](https://github.com/Automattic/pocket-casts-android/assets/95022986/c7bb38dc-48c2-4d4f-925b-b25ff8f81657) | ![select](https://github.com/Automattic/pocket-casts-android/assets/95022986/529a4f09-5b1c-4581-94a7-c7124a0ba120) 



## Checklist
 
#### I have tested any UI changes...
